src/config.js

```diff
@@ -0,0 +1,203 @@
+        MultisigWalletConfig.supportedNetworks.includes(network), 
+        `Network ${network} not supported.`
+      );
+      this.network = network;
+    } else {
+      this.network = MAINNET;
+    }
+    
+    if (client) {
+      assert(
+        MultisigWalletConfig.supportedClients.includes(client.type),
+        `Client type "${client.type}" not supported`
+      )
+      this.client = client;
+    }
+    
+    assert(typeof addressType === 'string', "Wallet config needs addressType.");
+    this.addressType = addressType;
+
+    assert(
+      extendedPublicKeys && 
+      Array.isArray(extendedPublicKeys),
+      "Wallet config needs array of extendedPublicKeys."
+    );
+    this.extendedPublicKeys = extendedPublicKeys;
+    this.validateExtendedPublicKeys(false);
+
+    assert(
+     requiredSigners && requiredSigners <= extendedPublicKeys.length,
+      "Wallet config needs requiredSigners."
+    );
+    this.requiredSigners = requiredSigners;
+    this.totalSigners = extendedPublicKeys.length;
+
+    if (derivation) {
+      const pathError = validateBIP32Path(derivation);
+      assert(!pathError.length, pathError);
+      this.derivation = derivation;
+    }
+
+    if (startingAddressIndex) {
+      if (typeof startingAddressIndex !== 'number') {
+        throw new TypeError('Expected int for startingAddressIndex');
+      }
+      this.startingAddressIndex = startingAddressIndex;
+    }
+  }
+
+  /**
+   * @description instantiate a new wallet config object from a JSON
+   * @param {string} jsonString - json string to convert.
+   * @returns {MultisigWalletConfig} returns a new instance of a MultisigWalletConfig
+   */
+  static fromJSON(jsonString) {
+    if (typeof jsonString !== 'string') throw new TypeError('Must pass a valid JSON string');
+    let options;
+
+    try {
+      options = JSON.parse(jsonString);
+    } catch (error) {
+      throw new Error("Unable to parse JSON.");
+    }
+    
+    // TODO: how do we handle caravan configs where the extended public keys
+    // don't include xfp information? We can get a parent fingerprint 
+    // from the xpub and use that. It wouldn't be valid as an xfp but could be used as stand in? 
+    return new this(options);
+  }
+
+  /**
+   * @description static method wrapper around getting a new
+   * config based on a JSON since Caravan is the assumed format of accepted JSON
+   * @param {string} jsonString - valid config json string from Caravan export
+   * @returns {MultisigWalletConfig} new instance of a MultisigWalletConfig
+   */
+  static fromCaravanConfig(jsonString) {
+    return MultisigWalletConfig.fromJSON(jsonString);
+  }
+
+  /**
+   * @description Validate the public keys on the config instances. 
+   * Not all configs require xfp in the xpub objects, so an optional param is
+   * accepted to dictate whether or not this value should be validated
+   * @param {boolean} requiresXFP - does the object require an xfp, if so it will also be validated
```

Comment:
this is a little confusing because you could require a *single* xfp - and then the rest can be noise. but the way this works now is if you say requiresXFP every xpub must have an xfp.

Reply:
I would say that those two statements are equivalent. i.e. 1 valid xpfs + 2 "noisy" xfps == every xpub has an xfp. Basically what this is saying that if a root fingerprint is required, make sure it's included. We can't validate that it's not noisy, but that's not really relevant in these operations since we can't really validate them anyway. So if it's required, make sure they all have _something_ and that it's valid hex, correct length, etc.

Reply:
I'm addressing this with an `addPlaceholderFingerprint` method on the class that can be used when we have at least one xfp (assumed valid) and need the rest to have a placeholder in there.